### ISSUE TITLE
Locale for String formatting

### ISSUE THREAD

**Issue by schildbach** [tony19/logback-android#87]
Whenever there is any String formatting we need a way to supply a Locale.

e.g. I use filePattern.setPattern("%d{HH:mm:ss.SSS} [%thread]"); and in the Arabic world this yields log lines like this:

١٨:٣٧:١١.٧٢٧ [main] …

A similar thing happens for the filenames of rolling logs:

rollingPolicy.setFileNamePattern("/%d.log.gz");

yields files named like this:

٢٠١٥-٠٢-١٦.-1847019678.log.gz

You may notice these filenames are writting from right to left partially. (-:

**Comment by tony19** [comment]
What are the expected values for your examples?

**Comment by schildbach** [comment]
Well, Latin numbers. Something like 2015-02-19 etc. If there were a locale, I'd probably pick Locale.US.

**Comment by tony19** [comment]
Are you saying the locale should be hard-coded to US? You mentioned partial right-to-left (and I'm not sure what the full line is supposed to look like), so I'm a little confused.

**Comment by schildbach** [comment]
No. What I meant is:

Whenever there is any String formatting we need a way to supply a locale.

I personally will set this locale to Locale.US for all of my apps. I can imagine devs from "non-latin-number countries" may decide different.

**Comment by schildbach** [comment]
Regarding RTL: try cut & paste (or even just text selection) on my first comment https://github.com/tony19/logback-android/issues/87#issue-58103285 , second to last line. You'll see it's mixed LTR/RTL.

**Comment by guohai** [comment]
I'm also facing this problem.
For me, I prefer timestamp in log is like this
<pre>
2015-07-13 16:48:25.823+0800 [Thread-128] DEBUG
</pre>
not like this
<pre>
٢٠١٥-٠٩-٠٣ ٠٩:٠٢:٤٢.٤٩٥+0400 [main] INFO
</pre>

**Comment by tony19** [comment]
The locale can now be specified in date conversion patterns in the third option position. By default, the US/English locale is assumed.

Examples:
```
  %d{HH:mm:ss,EST,fr}
  %d{HH:mm:ss,EST,"zh,CN"}
  %d{HH:mm:ss,EST,"hi,IN"}
```

**Comment by tony19** [comment]
Released in 1.1.1-5
